algorithms/kernel/svm/svm_predict_impl.i

```diff
@@ -43,6 +43,134 @@ using namespace daal::internal;
 using namespace daal::services;
 using namespace daal::services::internal;
 
+template <typename algorithmFPType, CpuType cpu>
+class PredictTask
+{
+public:
+    DAAL_NEW_DELETE();
+    virtual ~PredictTask() {}
+
+    bool isValid() const { return _buff.get(); }
+
+    services::Status kernelCompute(const size_t startRow, const size_t nRows)
+    {
+        services::Status status;
+        NumericTablePtr shResNT = HomogenNumericTableCPU<algorithmFPType, cpu>::create(_buff.get(), _nSV, nRows, &status);
+        DAAL_CHECK_STATUS_VAR(status);
+
+        auto xBlockNT = getBlockNTData(startRow, nRows, status);
+        DAAL_CHECK_STATUS_VAR(status);
+        _shRes->set(kernel_function::values, shResNT);
+        _kernel->getInput()->set(kernel_function::X, xBlockNT);
+        _kernel->getParameter()->computationMode = kernel_function::matrixMatrix;
+
+        return _kernel->computeNoThrow();
+    }
+
+    const algorithmFPType * getBuff() const { return _buff.get(); }
+
+protected:
+    PredictTask(const size_t nRowsPerBlock, const NumericTablePtr & xTable, const NumericTablePtr & svTable, kernel_function::KernelIfacePtr & kernel)
+        : _xTable(xTable), _nSV(svTable->getNumberOfRows()), _nFeatures(svTable->getNumberOfColumns())
+    {
+        services::Status status;
+        _buff.reset(_nSV * nRowsPerBlock);
+        _kernel = kernel->clone();
+        _shRes  = kernel_function::ResultPtr(new kernel_function::Result());
+        _kernel->setResult(_shRes);
+        _kernel->getInput()->set(kernel_function::Y, svTable);
+    }
+
+    virtual NumericTablePtr getBlockNTData(const size_t startRow, const size_t nRows, services::Status & status) = 0;
+
+protected:
+    const NumericTablePtr & _xTable;
+    const size_t _nSV;
+    const size_t _nFeatures;
+    TArrayScalable<algorithmFPType, cpu> _buff;
+    kernel_function::KernelIfacePtr _kernel;
+    kernel_function::ResultPtr _shRes;
+};
+
+template <typename algorithmFPType, CpuType cpu>
+class PredictTaskDense : PredictTask<algorithmFPType, cpu>
+{
+public:
+    using Super = PredictTask<algorithmFPType, cpu>;
+    DAAL_NEW_DELETE();
+    virtual ~PredictTaskDense() {}
+
+    static Super * create(const size_t nRowsPerBlock, const NumericTablePtr & xTable, const NumericTablePtr & svTable,
+                          kernel_function::KernelIfacePtr & kernel)
+    {
+        auto val = new PredictTaskDense(nRowsPerBlock, xTable, svTable, kernel);
+        if (val && val->isValid()) return val;
+        delete val;
+        return nullptr;
+    }
+
+protected:
+    PredictTaskDense(const size_t nRowsPerBlock, const NumericTablePtr & xTable, const NumericTablePtr & svTable,
+                     kernel_function::KernelIfacePtr & kernel)
+        : Super(nRowsPerBlock, xTable, svTable, kernel)
+    {}
+
+    NumericTablePtr getBlockNTData(const size_t startRow, const size_t nRows, services::Status & status) override
+    {
+        _xBlock.set(*Super::_xTable, startRow, nRows);
```

Comment:
You should check the block: ```DAAL_CHECK_BLOCK_STATUS_THR(_xBlock);```
And the same for CSR case

Reply:
I added check for status. But I don't use `DAAL_CHECK_BLOCK_STATUS_THR`, because `getBlockNTData` return not status